File: js/나이트의 이동(7562) copy.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const T = Number(input[0]);
let index = 1;

const dx = [2, 1, 1, 2, -2, -1, -1, -2];
const dy = [1, 2, -2, -1, -1, -2, 2, 1];

for (let t=0; t<T; t++) {
    const I = Number(input[index]);
    index++;
    const [start_x, start_y] = input[index].split(' ').map(Number);
    index++;
    const [end_x, end_y] = input[index].split(' ').map(Number);
    index++;

    const graph = Array.from({length: I}, () => Array(I).fill(0));
    const visited = Array.from({length: I}, () => Array(I).fill(false));

    function bfs(x,y) {
        let queue = [];
        queue.push([x,y]);
        visited[x][y] = true;

        while (queue.length) {
            const [x,y] = queue.shift();

            if (x === end_x && y === end_y) {
                console.log(graph[x][y]);
                return;
            }

            for (let i=0; i<8; i++) {
                let nx = x + dx[i];
                let ny = y + dy[i];

                if (nx >=0 && nx < I && ny >=0 && ny < I && !visited[nx][ny] && graph[nx][ny] === 0) {
                    visited[nx][ny] = true;
                    graph[nx][ny] = graph[x][y] + 1;
                    queue.push([nx,ny]);
                }
            }
        }
    }

    bfs(start_x, start_y);

}

File: python/나이트의 이동(7562) copy.py
# 입력 첫째 줄 : tc 개수
# [ 각 테스트 케이스 ]
# 1번째 줄 : 한 변의 길이 I (I*I)
# 2번째 줄 : 현재 있는 칸
# 3번째 줄 : 이동하려는 칸

from collections import deque

dx = [2, 1, -1, -2, 2, 1, -1, -2]
dy = [1, 2, -2, -1, -1, -2, 2, 1]

tc = int(input())
for _ in range(tc):
    I = int(input())
    start_x, start_y = tuple(map(int, input().split()))
    end_x, end_y = tuple(map(int, input().split()))
    visited = [[False] * I for _ in range(I)]
    graph = [[0] * I for _ in range(I)]

    def bfs(x,y):
        queue = deque()
        queue.append((x,y))
        visited[x][y] = True

        while queue:
            x,y = queue.popleft()

            if x == end_x and y == end_y:
                print(graph[x][y])
                return
            
            for k in range(8):
                nx = x + dx[k]
                ny = y + dy[k]

                if nx >= 0 and nx < I and ny >=0 and ny < I and visited[nx][ny] == False and graph[nx][ny] == 0:
                    graph[nx][ny] = graph[x][y] + 1
                    queue.append((nx,ny))
                    visited[nx][ny] = True
    
    bfs(start_x, start_y)